frontend/src/index.js

```diff
@@ -1,29 +1,29 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { ReactKeycloakProvider } from '@react-keycloak/web';
 import { QueryClient, QueryClientProvider } from 'react-query';
-import keycloak from './keycloak';
 import AppRouter from './routes';
-
+import { KeycloakContext, keycloaks } from './keycloak';
 import './styles/index.scss';
 
 const queryClient = new QueryClient();
 
 const initOptions = {
   onLoad: 'check-sso',
-  pkceMethod: 'S256',
-  useNonce: false
+  pkceMethod: 'S256'
 };
 
-ReactDOM.render(
-  <ReactKeycloakProvider
-    authClient={keycloak}
-    initOptions={initOptions}
-    LoadingComponent={<div>Loading</div>}
-  >
-    <QueryClientProvider client={queryClient}>
-      <AppRouter />
-    </QueryClientProvider>
-  </ReactKeycloakProvider>,
-  document.getElementById('root')
-);
+const keycloakPromises = [];
+for (const i in keycloaks) {
+  keycloakPromises.push(keycloaks[i].init(initOptions));
+}
+
+Promise.all(keycloakPromises).then(() => {
```

Comment:
If we could somehow expose an initialized attribute like the previous hook (set true when both keycloaks are intialized) I think it would be better to have that and then still have some sort of support for a LoadingComponent prop.

Reply:
`initOptions` could also then remain a prop, then the authClient prop would become an array of these two keycloak instances.

Reply:
https://github.com/react-keycloak/react-keycloak/blob/d4618984de713dc23b790d3ec6f7fa08f93b13dc/packages/core/src/provider.tsx#L79 This is what ReactKeycloakProvider manages state wise

Reply:
I'm not sure about this... I didn't use the ReactKeycloakProvider component because I don't think it can take multiple keycloaks in its authClient prop; or can it?

Reply:
Oh okay I see your comment about the custom provider; will look into it!